hw2/01-coins.js

```diff
@@ -1,7 +1,84 @@
+};
+
+//Function that will take a dollar amount as input and return its change broken down by coins.
 const calculateChange = (input) => {
   // Add your code here
+  //Multiple use cases to check before actual functionality.
+  if (input > 10) return 'Error: the number is too large.';
+  else if (input === 0) return 'Error: 0 returns no change.';
+  else if (input < 0) return 'Error: Negative dollar amount not allowed.';
+  else if (isNaN(input)) return 'Error: Input is not a number value.';
+  else {
+    //I decided to fix the values to be up to two decimal places.
+    //I was getting the wrong change amounts with regular float values.
+    let change = input.toFixed(2);
+    //Get the number of dollars by rounding down.
+    let dollars = Math.floor(input);
+    //Calculate a new value for the remaining change.
+    change = (change % 1).toFixed(2);
+    let quarters = Math.floor(change / 0.25);
+    change = (change % 0.25).toFixed(2);
+    let dimes = Math.floor(change / 0.1);
+    change = (change % 0.1).toFixed(2);
+    let nickels = Math.floor(change / 0.05);
+    change = (change % 0.05).toFixed(2);
+    let pennies = Math.floor(change / 0.01);
+
+    //Start the main message we will be returining.
+    let mainMessage = '$' + input.toFixed(2) + ' ==> ';
+
+    //Check each change type, if it is greater than 0, we want to add it to our string.
+    if (dollars > 0) {
+      //Use plurals() to get either "dollar" or "dollars"
+      let sDollars = plurals(dollars, 'dollar');
+      //For formatting purposes, we want to check if all the other change values are zero.
+      //If they are 0, we want to only append the dollar string to main message and return it.
+      if (quarters === 0 && dimes === 0 && nickels === 0 && pennies === 0)
+        return (mainMessage += dollars + ' ' + sDollars);
+      else mainMessage += dollars + ' ' + sDollars + ', ';
+      //Else, we will append the dollar string to main message and add a ", " at the end.
+    }
+    //The rest of these conditions essentially do the same thing.
+    if (quarters > 0) {
+      let sQuarters = plurals(quarters, 'quarter');
+      if (dimes === 0 && nickels === 0 && pennies === 0)
+        return (mainMessage += quarters + ' ' + sQuarters);
+      else mainMessage += quarters + ' ' + sQuarters + ', ';
+    }
+    if (dimes > 0) {
+      let sDimes = plurals(dimes, 'dime');
+      if (nickels === 0 && pennies === 0)
+        return (mainMessage += dimes + ' ' + sDimes);
+      else mainMessage += dimes + ' ' + sDimes + ', ';
+    }
+    if (nickels > 0) {
+      let sNickels = plurals(nickels, 'nickel');
+      if (pennies === 0) return (mainMessage += nickels + ' ' + sNickels);
+      else mainMessage += nickels + ' ' + sNickels + ', ';
+    }
+    if (pennies > 0) {
+      let sPennies = plurals(pennies, 'penny');
+      mainMessage += pennies + ' ' + sPennies;
+      return mainMessage;
+    }
+  }
 };
```

Comment:
This all makes a lot of sense, and it is very clear. However, I feel like you're repeating a lot and it could be cleaned up a bit.  I had used a multi dimensional array, and while I think they're not the easiest to use when programming. it cleaned up a lot of repetition.

Also, [6.3](https://airbnb.io/javascript/#es6-template-literals) indicates the preference of template strings. But I'm indifferent with this style preference.